### PR TITLE
refactor: create IPC_MESSAGES enum for IPC message channels

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Now that almost everything is converted to TypeScript it's quite safe to do this. Eliminates duplicate string constants for internal IPC channels.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: no-notes

### CODE REVIEW

File: lib/browser/guest-view-manager.js [hunk 1]
```diff
@@ -115,30 +115,30 @@ const createGuest = function (embedder, params) {
   // Dispatch events to embedder.
   const fn = function (event) {
     guest.on(event, function (_, ...args) {
-      sendToEmbedder('ELECTRON_GUEST_VIEW_INTERNAL_DISPATCH_EVENT', event, ...args);
+      sendToEmbedder('GUEST_VIEW_INTERNAL_DISPATCH_EVENT', event, ...args);
```
Comment (nornagon) [inline lib/browser/guest-view-manager.js:118]: can these be enums too?

File: lib/renderer/web-view/guest-view-internal.ts [hunk 2]
```diff
@@ -66,18 +67,18 @@ const dispatchEvent = function (
 };
 
 export function registerEvents (webView: WebViewImpl, viewInstanceId: number) {
-  ipcRendererInternal.on(`ELECTRON_GUEST_VIEW_INTERNAL_DESTROY_GUEST-${viewInstanceId}`, function () {
+  ipcRendererInternal.on(`GUEST_VIEW_INTERNAL_DESTROY_GUEST-${viewInstanceId}`, function () {
```
Comment (nornagon) [inline lib/renderer/web-view/guest-view-internal.ts:70]: can we use the enum here?

Comment (nornagon) [reply lib/renderer/web-view/guest-view-internal.ts:70]: This seems like a potentially useful typescript feature coming in 4.1: https://devblogs.microsoft.com/typescript/announcing-typescript-4-1-beta/#template-literal-types

**Review by nornagon — CHANGES_REQUESTED** [review]
Nice stuff! Requesting changes for usage of the enum in a couple more places.

Can we change the type signature of `ipcRendererInternal.send` so it requires its first argument to be a member of the enum?

File: lib/common/remote/ipc-messages.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export const enum IPC_MESSAGES {
```
Comment (nornagon) [inline lib/common/remote/ipc-messages.ts:1]: Please also apply this change to https://github.com/electron/remote.

File: lib/renderer/web-view/guest-view-internal.ts (hunk 2 above)
Comment (miniak) [reply lib/renderer/web-view/guest-view-internal.ts:70]: forgot this one

File: lib/browser/guest-view-manager.js (hunk 1 above)
Comment (miniak) [reply lib/browser/guest-view-manager.js:118]: done in TypeScript, I don't think we can use `const enum` in JavaScript

Comment (miniak) [reply lib/browser/guest-view-manager.js:118]: they cannot be used here in JavaScript code, needs to be converted to TypeScript

Comment (miniak) [reply lib/browser/guest-view-manager.js:118]: or I can change the `enum` to be non-`const`

File: lib/common/remote/ipc-messages.ts (hunk 3 above)
Comment (miniak) [reply lib/common/remote/ipc-messages.ts:1]: https://github.com/electron/remote/pull/10

**Comment by miniak** [comment]
> Can we change the type signature of ipcRendererInternal.send so it requires its first argument to be a member of the enum?
@nornagon that's more complicated and might require converting the last 2 files to TypeScript.

**Review by alexeykuzmin — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

File: lib/browser/remote/server.ts [hunk 4]
```diff
@@ -30,7 +31,7 @@ const finalizationRegistry = new FinalizationRegistry((fi: FinalizerInfo) => {
   const ref = rendererFunctionCache.get(mapKey);
   if (ref !== undefined && ref.deref() === undefined) {
     rendererFunctionCache.delete(mapKey);
-    if (!fi.webContents.isDestroyed()) { fi.webContents.sendToFrame(fi.frameId, 'ELECTRON_RENDERER_RELEASE_CALLBACK', fi.id[0], fi.id[1]); }
+    if (!fi.webContents.isDestroyed()) { fi.webContents.sendToFrame(fi.frameId, 'RENDERER_RELEASE_CALLBACK', fi.id[0], fi.id[1]); }
```
Comment (nornagon) [inline lib/browser/remote/server.ts:34]:
```suggestion
    if (!fi.webContents.isDestroyed()) { fi.webContents.sendToFrame(fi.frameId, IPC_MESSAGES.RENDERER_RELEASE_CALLBACK, fi.id[0], fi.id[1]); }
```

**Review by nornagon — APPROVED** [review]

**Pull request merged**